### ISSUE TITLE
Incorrect IdentityBinaryExpression message?

### ISSUE THREAD

**Issue by axkr** [google/error-prone#3398]
errorprone version 2.15.0

For binary logical operations with multiple decrements on an array index like this

```java
if (((sieveArray[x--] | sieveArray[x--] | sieveArray[x--] | sieveArray[x--]) & 0x80) != 0) {
   ...

   ...
}
```

errorprone prints a `IdentityBinaryExpression` message:
```
[IdentityBinaryExpression] A binary expression where both operands are the same is usually incorrect; the value of this expression is equivalent to `sieveBlock[x--]`.
```

https://errorprone.info/bugpattern/IdentityBinaryExpression

The message is propably wrong?

**Comment by graememorgan** [comment]
That page does acknowledge possible false positives from side-effectful expressions:

> If the expression has side-effects, consider refactoring one of the expressions with side effects into a local.
